src/vcpkg/postbuildlint.cpp

```diff
@@ -554,11 +555,9 @@ namespace vcpkg::PostBuildLint
                                              const std::vector<Path>& files,
                                              const Filesystem& fs)
     {
-#if defined(_WIN32)
         std::vector<FileAndArch> binaries_with_invalid_architecture;
-
-        for (const Path& file : files)
-        {
+#if defined(_WIN32)
```

Comment:
IIUC, this check will be skipped if vcpkg is running on Linux/Mac host but the library is being built for windows (MinGW or something).

Reply:
+1 to this, this should really be based on the target triplet, not the OS that vcpkg-tool was built on.

We even are already doing this below w.r.t. the `lib_filter`